### PR TITLE
Use space and double click to pause/unpause credits scroll.

### PR DESCRIPTION
**Pull request by quyykk**
## Feature Details

Allows the user to pause/unpause the credits scroll by pressing space, which is a pretty natural hotkey I think. I remember other games where this is also the case, but don't remember which ones. 

Also allows double clicking the credits to pause/unpause as well.

## Testing Done

Paused and unpaused the credits.

### CODE REVIEW

**Comment by tehhowch** [comment]
Another intuitive scroll pause/unpause trigger is to click/double-click on the scrollable area. (I'd lean towards double-click, in case we support text selection someday.)

File: source/MenuPanel.cpp [hunk 1]
```diff
@@ -179,8 +178,25 @@ bool MenuPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
...
 
 	return true;
 }
+
+
+
+bool MenuPanel::Click(int x, int y, int clicks)
+{
+	// Double clicking on the credits pauses/resumes the credits scroll.
+	auto creditsBox = Rectangle::WithCorners(Point(-490., -165.), Point(-230., 115.));
```
[the remark this answers is not in the document]
Comment (quyykk) [reply source/MenuPanel.cpp]: Might as well do it in this PR.

**Review by tehhowch — APPROVED** [review]
LGTM, deferring merge to another maintainer

**Comment by petervdmeer** [comment]
I also briefly reviewed; looks good!

**Pull request merged**